starboard/nplb/compiler_compliance/cpp17_support.cc

```diff
@@ -64,6 +64,12 @@ static_assert(add_one(1) == 2, "Constexpr lambdas support is required");
 static_assert(starboard::foo::bar::baz::life() == 42,
               "Nested Namespaces support is required");
 
+// From https://gcc.gnu.org/projects/cxx-status.html#cxx17
+// test template argument deduction for class templates
+#if __cpp_deduction_guides < 201611
```

Comment:
This seems fine and by the spec. However, to make this more of real a compiler check i'd add in actual stub code that relies on template argument deduction here.

There are no hard guarantees that all compiler vendors that we have to deal with properly manifest the supported features - i'd rather actually test the compiler behavior.

Reply:
Also please have a look at language features expected to be currently required by Chrome : https://chromium.googlesource.com/chromium/src/+/main/styleguide/c++/c++-features.md#core-allowlist-17

Of course, this list is always evolving in trunk.